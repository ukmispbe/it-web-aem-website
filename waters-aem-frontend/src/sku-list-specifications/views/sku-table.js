import React from 'react';
import PropTypes from 'prop-types';
import SkuTableItem from './sku-table-item';
import { callCustomerPriceApi } from '../../utils/userFunctions';
import LoginStatus from '../../scripts/loginStatus';
import CheckOutStatus from '../../scripts/checkOutStatus';
import Ecommerce from '../../scripts/ecommerce';

class SkuTable extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            config: this.props.config,
            skuInfo: this.props.config.skuInfo,
            userInfo: callCustomerPriceApi(
                this.props.config.isCustomerPriceApiDisabled
            ),
        };
    }
    getSkuListItems = (skuList) =>
        skuList.map((item) => (
            <SkuTableItem
                skuData={item}
                userInfo={this.state.userInfo}
                config={this.props.config}
                checkEcommCondition={this.checkEcommCondition}
            />
        ));

    checkEcommCondition = () => {
        if (Ecommerce.isDisabledState()) {
            return false;
        } else if (
            (Ecommerce.isPartialState() &&
                LoginStatus.state() &&
                CheckOutStatus.state()) ||
            (!Ecommerce.isPartialState() && !Ecommerce.isDisabledState())
        ) {
            return true;
        }
        return false;
    };
    render() {
        const { config, skuList } = this.props;
        return (
            <section class="sku-table">
                <section class="sku-table__header row">
                    <span className="item col-lg col-lg-1">
                        {config.skuNumberLabel}
                    </span>
                    <span className="item col-lg">{config.productLabel}</span>
                    {this.checkEcommCondition() ? (
                        <>
                            <span className="item col-lg col-lg-2">
                                {config.availabilityLabel}
                            </span>
                            <span className="item col-lg col-lg-1">
                                {config && config.skuInfo
                                    ? config.skuInfo.listPriceLabel
                                    : ''}
                            </span>
                            {LoginStatus.state() && (
                                <span className="item col-lg col-lg-1 regular">
                                    {config && config.skuInfo
                                        ? config.skuInfo.custPriceLabel
                                        : ''}
                                </span>
                            )}
                            <span className="item col-lg col-lg-2"></span>
                        </>
                    ) : (
                        <span className="item col-lg col-lg-6"></span>
                    )}
                </section>
                <ul class="sku-table__body">
                    {skuList && this.getSkuListItems(skuList)}
                </ul>
            </section>
        );
    }
}

SkuTable.propTypes = {
    config: PropTypes.object.isRequired,
    skuList: PropTypes.object,
};

SkuTable.defaultProps = {
    config: {},
    skuList: [],
};

export default SkuTable;
