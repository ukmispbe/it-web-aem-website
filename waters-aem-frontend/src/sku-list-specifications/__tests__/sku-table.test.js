import React from 'react';
import { shallow } from 'enzyme';

//Mocked Props
import SkuTable from '../views/sku-table';
import { props, skuListData } from '../sku-table.mock';


describe('<SkuTable />', () => {
    let enzymeWrapper;
    
    beforeAll(() => {        
        enzymeWrapper = shallow(<SkuTable {...props} />);
    });
    it('should render without throwing an error', () => {
        expect(enzymeWrapper).toMatchSnapshot();
    }); 
    
    it('should render with sku data ', () => {
        const wrapper = shallow(<SkuTable {...skuListData} />)
        expect(wrapper).toMatchSnapshot();
    }); 
});