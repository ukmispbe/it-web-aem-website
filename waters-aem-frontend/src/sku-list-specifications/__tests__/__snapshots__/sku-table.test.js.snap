// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkuTable /> should render with sku data  1`] = `ShallowWrapper {}`;

exports[`<SkuTable /> should render without throwing an error 1`] = `ShallowWrapper {}`;
